mqtt/handler.go

```diff
@@ -162,32 +162,32 @@ func (h *handler) Publish(c *session.Client, topic *string, payload *[]byte) {
 	// Topics are in the format:
 	// channels/<channel_id>/messages/<subtopic>/.../ct/<content_type>
 
-	channelParts := channelRegExp.FindStringSubmatch(*topic)
-	if len(channelParts) < 2 {
+	subtopicParts := subtopicRegExp.FindStringSubmatch(*topic)
```

Comment:
Create a method `validateSubtopic`, move it to the messaging package and use it in the rest of the code where we validate the subtopic including any other adapter.